src/util/cpplibostree.hpp

```diff
@@ -115,8 +115,76 @@ class OSTreeRepo {
      * @return true if the commit is signed
      * @return false if the commit is not signed
      */
-    static bool isCommitSigned(const Commit& commit);
+    static bool IsCommitSigned(const Commit& commit);
+
+    // read & write access to OSTree repo:
 
+    /**
+     * @brief Promotes a commit to another branch. Similar to:
+     * `ostree commit --repo=repo -b newRef -s newSubject --tree=ref=hash`
+     *
+     * @param hash hash of the commit to promote
+     * @param newRef branch to promote to
+     * @param addMetadataStrings list of metadata strings to add -> KEY=VALUE
+     * @param newSubject new commit subject, it needed
+     * @param keepMetadata should new commit keep metadata of old commit
+     * @return true on success, false on failed promotion
+     */
+    bool PromoteCommit(const std::string& hash,
+                       const std::string& newRef,
+                       const std::vector<std::string> addMetadataStrings,
+                       const std::string& newSubject = "",
+                       bool keepMetadata = true);
+
+    /**
+     * @brief Removes a commit (and all its predecessors, if they would)
+     *
+     * Synonym to:
+     *  [ `ostree reset --repo=<repo> <ref> <ref>^` ]
+     *  `ostree prune --repo=<repo> --delete-commit=<hash>`
+     *
+     * Effect:
+     *  If the commit is the most recent on its branch -> reset head of branch.
+     *  If not, then remove this commit and all predecessors (that would otherwise be unreachable).
+     *
+     * @param commit Commit to remove (must match an element in the `GetCommitList()`).
+     * @return True on success.
+     */
+    bool RemoveCommitFromBranchAndPrune(const Commit& commit);
+
+    /**
+     * @brief Resets the specified branch head by one commit, similar to `git reset HEAD~`
+     *
+     * @param branch Branch to reset.
+     * @return True on success.
+     */
+    bool ResetBranchHeadAndPrune(const std::string& branch);
+
+    /**
+     * @brief Checks if commit is the most recent commit on its branch
+     *
+     * @param branch Branch to get most recent commit from.
+     * @return Most recent commit of the specified branch.
+     */
+    const Commit& GetMostRecentCommitOfBranch(const std::string& branch) const;
```

Comment:
```suggestion
    [[nodiscard]] const Commit& GetMostRecentCommitOfBranch(const std::string& branch) const;
```

For me the description does not make sense. As I understand it: "Returns the latest commit for the given branch name."

If this is true `[[nodiscard]]` makes sense since a user not dealing with the return value makes no sense.

Reply:
Yes, I'll go through the code and check where it makes sense, also in other places